src/ansiblelint/utils.py

```diff
@@ -521,12 +521,36 @@ def _sanitize_task(task: Dict[str, Any]) -> Dict[str, Any]:
     return result
 
 
+def _extract_ansible_parsed_keys_from_task(task: Dict[str, Any]) -> Dict[str, Any]:
+    """Return a dict with existing key in task."""
+    result = {}
+    for (k, v) in list(task.items()):
+        if k in ("action", "local_action", "args", "delegate_to"):
+            # we don't want to re-assign these values, which were
+            # determined by the ModuleArgsParser() above
+            continue
+        result[k] = v
+
+    return result
```

Comment:
Now that you've extracted this little utility, can we also use it for the action tasks (after the `is_nested_task` block)? Maybe one of the args could be the tuple of keys to extract. Wdyt?

Reply:
Thank you! I see. How about this?: [f0c431b](https://github.com/ansible/ansible-lint/pull/2031/commits/f0c431b76e941470fe211c246320cc418053eeda)

Reply:
Does this change make sense? or should I update the name or args?